pkg/segment/aggregations/evalaggs.go

```diff
@@ -23,297 +23,543 @@ import (
-				edgeValue = eValFloat
+			result, err := PerformEvalAggForMinOrMax(measureAgg, exists, enclosure, fieldToValue, isMin)
+			if err != nil {
+				return fmt.Errorf("ComputeAggEvalForMinOrMax: Error while performing eval agg for min/max, err: %v", err)
 			}
+			measureResults[measureAgg.String()] = result
 		}
 	}
-	enclosure, exists := measureResults[measureAgg.String()]
-	if !exists {
 
-		cVal := -1.7976931348623157e+308
-		if isMin {
-			cVal = math.MaxFloat64
-		}
+	return nil
+}
+
+func UpdateRangeStat(floatValue float64, rangeStat *structs.RangeStat) {
+	if floatValue < rangeStat.Min {
+		rangeStat.Min = floatValue
+	}
+	if floatValue > rangeStat.Max {
+		rangeStat.Max = floatValue
+	}
+}
 
-		enclosure = utils.CValueEnclosure{
-			Dtype: utils.SS_DT_FLOAT,
-			CVal:  cVal,
+func PerformEvalAggForRange(measureAgg *structs.MeasureAggregator, exists bool, currRangeStat structs.RangeStat, fieldToValue map[string]utils.CValueEnclosure) (structs.RangeStat, error) {
+	fields := measureAgg.ValueColRequest.GetFields()
+	finalRangeStat := structs.RangeStat{
+		Min: math.MaxFloat64,
+		Max: -math.MaxFloat64,
+	}
+	if len(fields) == 0 {
+		floatValue, _, isNumeric, err := GetFloatValueAfterEvaluation(measureAgg, fieldToValue)
+		// We cannot compute if constant is not numeric
+		if err != nil || !isNumeric {
+			return currRangeStat, fmt.Errorf("PerformEvalAggForRange: Error while evaluating value col request to a numeric value, err: %v", err)
+		}
+		UpdateRangeStat(floatValue, &finalRangeStat)
+	} else {
+		if measureAgg.ValueColRequest.BooleanExpr != nil {
+			boolResult, err := measureAgg.ValueColRequest.BooleanExpr.Evaluate(fieldToValue)
+			if err != nil {
+				return currRangeStat, fmt.Errorf("PerformEvalAggForRange: there are some errors in the eval function that is inside the range function: %v", err)
+			}
+			if boolResult {
+				finalRangeStat.Max = 1
+				finalRangeStat.Min = 1
+			}
+		} else {
+			floatValue, _, isNumeric, err := GetFloatValueAfterEvaluation(measureAgg, fieldToValue)
+			if err != nil {
+				return currRangeStat, fmt.Errorf("PerformEvalAggForRange: Error while evaluating value col request, err: %v", err)
+			}
+			// records that are not float will be ignored
+			if isNumeric {
+				UpdateRangeStat(floatValue, &finalRangeStat)
+			}
 		}
-		measureResults[measureAgg.String()] = enclosure
 	}
 
-	eValFloat, err := enclosure.GetFloatValue()
-	if err != nil {
-		return fmt.Errorf("ComputeAggEvalForMinOrMax: Attempted to perform aggregate min(), but the column %s is not a float value", fields[0])
+	if !exists {
+		return finalRangeStat, nil
 	}
 
-	if (isMin && eValFloat > edgeValue) || (!isMin && eValFloat < edgeValue) {
-		enclosure.CVal = edgeValue
-		measureResults[measureAgg.String()] = enclosure
+	if finalRangeStat.Min > currRangeStat.Min {
+		finalRangeStat.Min = currRangeStat.Min
 	}
-	return nil
+	if finalRangeStat.Max < currRangeStat.Max {
+		finalRangeStat.Max = currRangeStat.Max
+	}
```

Comment:
Ideally, we would only like to extend the range by comparing these relative values, because say if currRangeStat is not set then we could overwrite the finalRangeStat.Max with maxFloat. So, I think this is safe and is more readable.